plugins/plugin-chart-echarts/src/Pie/EchartsPie.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, createRef } from 'react';
+import styled from '@superset-ui/style';
+import echarts from 'echarts';
+
+interface EchartsPieStylesProps {
+  height: number;
+  width: number;
+}
+
+export type EchartsPiePropsReceived = {
+  height: number;
+  width: number;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: Record<any, any>; // please add additional typing for your data here
+  // add typing here for the props you pass in from transformProps.ts!
+  echartOptions: echarts.EChartOption;
+};
+
+// The following Styles component is a <div> element, which has been styled using Emotion
+// For docs, visit https://emotion.sh/docs/styled
+
+// Theming variables are provided for your use via a ThemeProvider
+// imported from @superset-ui/style. For variables available, please visit
+// https://github.com/apache-superset/superset-ui/blob/master/packages/superset-ui-style/src/index.ts
+
+const Styles = styled.div<EchartsPieStylesProps>`
+  height: ${({ height }) => height};
+  width: ${({ width }) => width};
+`;
+
+/**
+ * ******************* WHAT YOU CAN BUILD HERE *******************
+ *  In essence, a chart is given a few key ingredients to work with:
+ *  * Data: provided via `props.data`
+ *  * A DOM element
+ *  * FormData (your controls!) provided as props by transformProps.ts
+ */
+
+export default function EchartsPie(props: EchartsPiePropsReceived) {
+  // height and width are the height and width of the DOM element as it exists in the dashboard.
+  // There is also a `data` prop, which is, of course, your DATA 🎉
+  const { height, width, echartOptions } = props;
+
+  const rootElem = createRef<HTMLDivElement>();
+
+  // Often, you just want to get a hold of the DOM and go nuts.
+  // Here, you can do that with createRef, and the useEffect hook.
+  useEffect(() => {
+    const root = rootElem.current as HTMLDivElement;
+    // console.log('Plugin element', root);
+    const myChart = echarts.init(root);
+    myChart.setOption(echartOptions, true);
+    myChart.resize({
+      width,
+      height,
+    });
```

Comment:
I believe you can pass `width` and `height` to `init(...)`: https://echarts.apache.org/en/api.html#echarts.init

Reply:
The updated `width` and `height` still need to be passed to `resize()`, as otherwise it will get stuck in the original dimensions when resized. But I'll add them to `opts` in `init()` for clarity (perhaps it will make initial rendering faster when no resizing happens). Might also make sense to break out `init()` into an only once runnable `useEffect`.